docs/docs/how-to/plugins-and-themes/adding-gatsby-image-support.md

```diff
@@ -0,0 +1,206 @@
+
+## Adding a `gatsbyImageData` GraphQL resolver
+
+You can give your users the best experience by adding a `gatsbyImageData` resolver to your image nodes. This allows you to generate low-resolution or traced SVG images as inline data URIs, so your users can have blurred placeholders. You can also calculate the image's dominant color for an alternative placeholder. These are the same placeholders that are included with `gatsby-transformer-sharp`, and will give the best experience for your users. If you are able to deliver these directly from your CMS or other data source then this is ideal, but otherwise you can use helper functions included in `gatsby-plugin-sharp`.
+
+There are three steps to add a basic `gatsbyImageData` resolver:
+
+1. Create a `generateImageSource` function
+2. Create a resolver function
+3. Add the resolver
+
+### Create the `generateImageSource` function
+
+The `generateImageSource` function is where you define your image URLs. It is passed the base URL, width, height and format (i.e. the image filetytpe), as well as any custom options. You then return the generated URL. The returned object also includes width, height and format. This means you can return a different value from the one requested. For example, if the function requests an unsupported format or size, you can return a different one which will be used instead.
+
+```js:title=gatsby-source-example/gatsby-node.js
+// In this example we use a custom `quality` option
+const generateImageSource = (baseURL, width, height, format, fit, options) => {
+  const src = `https://myexampleimagehost.com/${baseURL}?w=${width}&h=${height}&fmt=${format}&q=${options.quality}`
+  return { src, width, height, format }
+}
+```
+
+### Create the resolver function
+
+You then can use the function created in the previous step to build your resolver function. It can be an async function, and it should return the value from `generateImageData` to An example resolver could look like this:
+
+```js::title=gatsby-source-example/gatsby-node.js
+
+import { generateImageData } from "gatsby-plugin-image"
+
+
+const resolveGatsbyImageData = async (image, options) => {
+   // The `image` argument is the node to which you are attaching the resolver,z
+   // so the values will depend on your data type.
+  const filename = image.src
+
+  const sourceMetadata = {
+    width: image.width,
+    height: image.height,
+    format: image.mimeType.split("/")[1]
+  }
+
+  // Generating placeholders is optional, but recommended
+  let placeholderURL
+
+  if(options.placeholder === "blurred") {
+    // This would be your own function to generate a low-resolution placeholder
+    placeholderURL: await getBase64Image({ filename })
+  }
+
+  // You could also calculate dominant color, and pass that as `backgroundColor`
+  // gatsby-plugin-sharp includes helpers that you can use to generate a tracedSVG or calculate
```

Comment:
I'm not exactly sure what these helpers are, maybe it could be possible to link to some docs/code (wherever it exists?)

Reply:
Yes, I need to document these!